# Find Cost of Tile to Cover W x H Floor

Calculate the total cost of tile it would take to cover a floor plan of width and height, using a cost entered by the user.

In [2]:
import timeit

In [25]:
import math

## Functions

Keywords only, no other functions.

In [38]:
def total_tile_cost(floor_width, floor_height, tile_width, tile_height):
    def validate_price():
        price = input('Give the tile price: $ ')
        while not price.isdigit():
            print('Error: only numbers allowed!')
            price = input('Give the tile price: $ ')
        return float(price)
    # TILE PRICE RECEIVES ONLY DIGITS
    tile_price = 1 #validate_price()
    # NUM OF HORIZONTAL TILES
    num_tiles_width = floor_width // tile_width
    if floor_width % tile_width != 0:
        num_tiles_width += 1
    # NUM OF VERTICAL TILES
    num_tiles_height = floor_height // tile_height
    if floor_height % tile_height != 0:
        num_tiles_height += 1
    
    return num_tiles_width * num_tiles_height * tile_price

Using .ceil()

In [39]:
def total_tile_cost2(floor_width, floor_height, tile_width, tile_height):
    def validate_price():
        price = input('Give the tile price: $ ')
        while not price.isdigit():
            print('Error: only numbers allowed!')
            price = input('Give the tile price: $ ')
        return float(price)
    # TILE PRICE RECEIVES ONLY DIGITS
    tile_price = 1 #validate_price()
    
    num_tiles_width = math.ceil(floor_width / tile_width)   
    num_tiles_height = math.ceil(floor_height / tile_height)

    return num_tiles_width * num_tiles_height * tile_price

Assigning .ceil to a local variable.

In [40]:
def total_tile_cost3(floor_width, floor_height, tile_width, tile_height):
    def validate_price():
        price = input('Give the tile price: $ ')
        while not price.isdigit():
            print('Error: only numbers allowed!')
            price = input('Give the tile price: $ ')
        return float(price)
    roundup = math.ceil
    # TILE PRICE RECEIVES ONLY DIGITS
    tile_price = 1 #validate_price()
    
    num_tiles_width = roundup(floor_width / tile_width)   
    num_tiles_height = roundup(floor_height / tile_height)

    return num_tiles_width * num_tiles_height * tile_price

## Tests

In [4]:
# (floor_width, floor_height, tile_width, tile_height)
dataset1 = (10, 10, 1, 1)
dataset2 = (10, 10, 1, 1.5)

In [21]:
total_tile_cost(*dataset2)

70.0

In [23]:
total_tile_cost2(*dataset2)

70

## Timeit
Disclaimer: set tile_price to a constant value for heavier loads of testing

In [41]:
%%timeit
total_tile_cost(*dataset1)

1.3 µs ± 12.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [42]:
%%timeit
total_tile_cost2(*dataset1)

1.46 µs ± 5.04 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [43]:
%%timeit
total_tile_cost3(*dataset1)

1.42 µs ± 8.52 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [19]:
1 - 6.12/16.8

0.6357142857142857

## Verdict
There's is an apparent 64% reduction in time when using math.ceil(), while assigning the .ceil() method to a local variable doesn't show performance improvement.